apps/passport-client/src/password.ts

```diff
@@ -9,3 +16,28 @@ export const checkPasswordStrength = (password: string): boolean => {
 };
 
 export const PASSWORD_MINIMUM_LENGTH = 8;
+
+// For when the user has not set a password yet, and wants to add one
+export const setPassword = async (password: string, dispatch: Dispatcher) => {
+  const crypto = await PCDCrypto.newInstance();
+  const currentEncryptionKey = loadEncryptionKey();
+  const { salt: newSalt, key: newEncryptionKey } =
```

Comment:
The logic from here down is the same as changing password, right?  Can the two codepaths be combined?  It would be nice if the complicated rekeying procedure was done in only one place.

Reply:
Will refactor out the common logic. One difference is that in the case of adding a new password, we'll have the load the encryption key from storage, whereas when we change a password, we ask for the current password to hash with the salt and generate the encryption key.